### DISCUSSION TITLE
Is there a metadata search similar to what the API provides?

### DISCUSSION THREAD

**Discussion by noancloarec** [thoughtspot/cs_tools#185] [Q&A] [answered]
Hi everyone,
We have developed a script to commit an org's changes to github using Toughtspot API V2, using 2 endpoints you provide
 - `/metadata/search` to search and filter the objects we want to commit 
 - `/vcs/git/branches/commit` to commit only a chosen set of objects

That works well, thank you for providing these features.

We'd now like to use the `cs_tools` CLI for the same purpose
The equivalent of `vcs/git/branches/commit` is `cs_tools tools git branches commit`. But I could not find an equivalent of `/metadata/search` . The closest thing I came up with is `cs_tools tools searchable metadata --org dev --syncer "json://directory=/home/noclo/Downloads/thoughtspot_search"`.
  
`/metadata/search` returns a rich metadata definition such as
```python
    {
        "metadata_id": "42705b19-e8dc-4181-9bbe-f90eadd6a54c",
        "metadata_name": "MY_TABLE", # I'm using this
        "metadata_type": "LOGICAL_TABLE",
        ...
        "metadata_header": {
            ...
            "authorName": "noan.cloarec",
            "authorDisplayName": "Noan Cloarec", # And these, so a user can know if an object belongs to them before commiting it
            ...
        },
    },
```

But cs_tools spreads its data into many files

```python
# ts_metadata_object.json
{
    "cluster_guid": "b7586e96-f587-11ee-9fba-5b9d2",
    ...
    "object_guid": "42705b19-e8dc-4181-9bbe-f90eadd6a54c",
    "name": "MY_TABLE",
    "author_guid": "c0c7b8dd-d858-42bb-b5ed-04cafbbe24ad", # No other info in this file
    ...
  },
# ts_user.json
  {
    "cluster_guid": "b7586e96-f587-11ee-9fba-5b9d2",
    "user_guid": "c0c7b8dd-d858-42bb-b5ed-04cafbbe24ad",
    "username": "noan.cloarec",
    "email": "my_email", 
    "display_name": "Noan Cloarec", # I'd need to join the 2 files to retrieve this field, and doing it in bash frightens me 😱
    ...
}
```
Also it does not seem to export worksheet, liveboard and answers data.
Is there a cs_tools metadata search closer to what the endpoint `/metadata/search` provides?

**Comment by boonhapus** [comment, marked as answer]
Unfortunately no, @noancloarec . If `metadata/search` gives you what you need, why are you trying to use CS Tools? What's ultimately your goal here?

This project wraps the same thoughtspot APIs and attempts to make them more consumable for non-developers. (We're still on V1 APIs but hope to change that soon.)

**Reply by noancloarec** [reply]
We're developing a CLI to ease comitting and github. Metadata search is used prior to commit to chose what should be committed.  
We thought cs_tools would allow us to implement these functionalities more easily than with a python script using your API.
We're developers so we will develop the CLI, thank you for your reply.

**Reply by boonhapus** [reply]
Makes sense @noancloarec . We're a pretty thin wrapper around the git apis today. In our tool, we use metadata/search as well, but purely to grab objects based on or type. We keep it fairly limited because otherwise the scope gets pretty large.

I've thought about splitting it off into a much smaller package recently and creating an action for GH Marketplace. I'll look into it again!
